src/main/java/ch/securify/analysis/AbstractDataflow.java

```diff
@@ -468,6 +468,8 @@ protected void deriveAssignTypePredicates() {
                 // TODO: double check whether to propagate the type of the
                 // argument to the output of blockhash
                 createAssignVarRule(instr, instr.getOutput()[0], instr.getInput()[0]);
+            } else if (instr instanceof ReturnDataCopy) {
+                // TODO: New memory-based rule here
```

Comment:
@ritzdorf @ptsankov I'd like to understand the implication of having this kind of incompleteness in this `deriveAssignTypePredicates`, does it just mean that the analysis will be less precise? Why is soundness still guarantees even with this `TODO`?